src/plugins/controller/UnicyclePID/UnicyclePID.cpp

```diff
@@ -93,33 +94,39 @@ void UnicyclePID::init(std::map<std::string, std::string> &params) {
     if (!speed_pid_.init(params["speed_pid"], false)) {
         std::cout << "Failed to set speed PID" << std::endl;
     }
+
+    parent_->state() = std::make_shared<State>();
+    *(parent_->state()) = *(parent_->state_truth());
 }
 
 bool UnicyclePID::step(double t, double dt) {
+    // Make a copy of the current state
+    StateWithCovariance ns(*(parent_->state()));
+
     heading_pid_.set_setpoint(vars_.input(desired_heading_idx_));
-    vars_.output(turn_rate_idx_, heading_pid_.step(time_->dt(), state_->quat().yaw()));
```

Comment:
I think we should still be using "state_" here, as well as the other places in this file.

Reply:
Done.